entity-service-impl/src/main/java/org/hypertrace/entity/query/service/converter/ValueHelper.java

```diff
@@ -155,10 +159,11 @@ public ConstantExpression convertToConstantExpression(final Value value)
     }
   }
 
-  public SubDocumentValue convertToSubDocumentValue(final Value value) throws ConversionException {
+  public SubDocumentValue convertToSubDocumentValue(
+      final AttributeKind attributeKind, final Value value) throws ConversionException {
     final ValueType type = value.getValueType();
 
-    if (isArray(type)) {
+    if (attributeMapping.isArray(attributeKind)) {
```

Comment:
I think this should continue to be based on the value type, because we are only dealing with values here. And, one can add (append) a primitive value to a list column. I think, the changes in this method can be reverted.